### DISCUSSION TITLE
Store mapping against Bluetooth ID rather than Device Connection order

### DISCUSSION THREAD

**Discussion by vaderag** [darthcloud/BlueRetro#50] [Ideas]
At present if you're using different controllers then you need to ensure you always connect them in the same order to get the mapping - mapping is assigned to device number as opposed to device

Would be great if instead mapping could be stored against the Bluetooth device uuid so that devices can be connected in any order yet have the correct mapping?

Example use case:
PowerA Gamecube controller for gamecube (as player one) is usual usage as it reflects the normal console
However, some ports you may want to use 8bitdo SN30 for this, or local emulation, or games more suited to SNES controller (e.g. Zelda Four Swords originally designed for GBA control)
Mapping will be different for both controllers, yet no way to do this currently

**Comment by darthcloud** [comment]
I think the real solution here is to improve the generic HID controller buttons mapping heuristics to detect a few common physical buttons layouts. I got an SF30 already and ordered a PowerA GC. I'll look onto it later this summer.

**Comment by darthcloud** [comment]
[v1.0-beta](https://github.com/darthcloud/BlueRetro/releases/tag/v1.0-beta)  fixup the face buttons rotation on PowerA GC and also implement a workaround to the powerA stopping to send report after inquiry mode stop.

All generic controller default mapping should be better now.
